### PR TITLE
feat(explorer): add social features and other improvements

### LINKED ISSUE #579: feat(explorer): add social integrations
The explorer could have a share on X button to simplify sharing that a proof has been verified on a batch.

### PR DESCRIPTION
**Pull request by glpecile**
> [!CAUTION]
> The `PHX_HOST` should be updated in the env file when deploying these changes.

# Changes

* improve [previous lighthouse score](https://pagespeed.web.dev/analysis/https-stage-alignedlayer-com/1pshfyev69?form_factor=desktop)
  * add webp version of images
  * add aria label to dark mode button and titles to its SVGs
* add post on X button to batch detail using the X api
  * disclaimer: when using an adblock the explorer link isn't displayed in the tweet to publish.
* add `@impl true` for better compiler validation on phoenix LV files
* lift functions up from render as it is a good practice
* make functions unused in other places private
* refactor difficult to read code
* add error views

## Test

The PR is currently deployed on [stage](https://stage.alignedlayer.com/).

### Before

image

### After

image

image

### Before

image

### After

image

### CODE REVIEW

**Review by uri-99 — APPROVED** [review]

File: explorer/.env.dev [hunk 1]
```diff
@@ -14,3 +14,6 @@ DB_HOST=localhost
 
 # Config file
 ALIGNED_CONFIG_FILE="../contracts/script/output/devnet/alignedlayer_deployment_output.json"
+
+# Debug
+DEBUG_ERRORS=true
```
Comment (JuArce) [inline explorer/.env.dev]: Missing new line

File: explorer/.env.example [hunk 2]
```diff
@@ -14,3 +14,6 @@ DB_HOST=<db_host>
 
 # Config file
 ALIGNED_CONFIG_FILE="<aligned_config_file>"
+
+# Debug
+DEBUG_ERRORS=<true|false>
```
Comment (JuArce) [inline explorer/.env.example]: Missing new line

File: explorer/.env.dev (hunk 1 above)
Comment (glpecile) [inline explorer/.env.dev]:
```suggestion
DEBUG_ERRORS=true

```

File: explorer/.env.example (hunk 2 above)
Comment (glpecile) [inline explorer/.env.example]:
```suggestion
DEBUG_ERRORS=<true|false>

```

**Review by JuArce — APPROVED** [review]

**Pull request merged**